### PR TITLE
Upgrade to openapi v3 and fixes 2019.10 specification

### PR DESCRIPTION
**Pull request by colin-swu**
## Description

* Upgrades the generation to openapi version 3.
* Fixes a parameter in /templates and /templates/:id paths (drafts previously was provided).
* Adds tests using GitHub actions.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/IO.Dyspatch.Tests/IntegrationTest.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+using System;
+using Xunit;
```
Comment (dlsteuer) [inline src/IO.Dyspatch.Tests/IntegrationTest.cs:2]: personally not a huge fan of xunit, i always preferred nunit but not a big deal

File: src/IO.Dyspatch.Tests/IntegrationTest.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        [Fact]
+        public void GetTemplates()
+        {
+            var accept = "application/vnd.dyspatch.2019.10+json";
+            var id = "tem_01de5teh6k59kya8q92mb01qzq";
+            var lang = "handlebars";
+
+            templates.GetTemplates(accept, "");
+            templates.GetTemplateById(id, lang, accept);
```
Comment (dlsteuer) [inline src/IO.Dyspatch.Tests/IntegrationTest.cs:36]: should there be some sort of assertion at least that we don't a 200 back?

File: src/IO.Dyspatch.Tests/IntegrationTest.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        [Fact]
+        public void GetDrafts()
+        {
+            var accept = "application/vnd.dyspatch.2019.10+json";
+            var id = "tdft_01dxkwr0nevs5h2baa3n3dgktp";
+            var lang = "handlebars";
+
+            drafts.GetDrafts(accept, "");
+            drafts.GetDraftById(id, lang, accept);
```
Comment (dlsteuer) [inline src/IO.Dyspatch.Tests/IntegrationTest.cs:47]: should there be an assertion here as well?

File: src/IO.Dyspatch.Tests/IntegrationTest.cs (hunk 1 above)
Comment (colin-swu) [reply src/IO.Dyspatch.Tests/IntegrationTest.cs:2]: Lol I'm just happy I got some sort of unit.

File: src/IO.Dyspatch.Tests/IntegrationTest.cs (hunk 2 above)
Comment (colin-swu) [reply src/IO.Dyspatch.Tests/IntegrationTest.cs:36]: I believe the client library will throw an exception if it fails, but we could add an additional assertion. What would you suggest?

Comment (dlsteuer) [reply src/IO.Dyspatch.Tests/IntegrationTest.cs:36]: Ok you are right, for any status code >= 400 it throws an Api Exception that works for me for now.

**Review by dlsteuer — APPROVED** [review]

**Pull request merged**